dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt

```diff
@@ -144,7 +137,9 @@ internal class CoreFeature(
     internal var site: DatadogSite = DatadogSite.US1
 
     internal lateinit var uploadExecutorService: ScheduledThreadPoolExecutor
-    internal lateinit var persistenceExecutorService: ExecutorService
+    internal lateinit var persistenceExecutorService: FlushableExecutorService
```

Comment:
Now that I see this problem I wonder if we should not use the factories for the features persistence + uploader executors. This way each feature will have its own. We will exchange more threads with less backpressure but I think this will be more performant as each feature has its own persistence folder so everything will be faster.

Reply:
right now the persistence is using the factory, as for the uploader, so far it is blocked because we are tightly coupled to the implementation class but we can take another go at this in the future.